src/ucm/configure.m4

```diff
@@ -4,8 +4,24 @@
 # See file LICENSE for terms.
 #
 
-AC_SUBST([UCM_MODULE_LDFLAGS],
-         ["-Xlinker -z -Xlinker interpose -Xlinker --no-as-needed"])
+AS_CASE([$host_os],
+    [linux*], [
```

Comment:
no need to check host os. if AC_LINK_IFELSE -s ok, regardless of OS, we can use those flags